mlprimitives/candidates/dsp.py

```diff
@@ -0,0 +1,211 @@
+
+        """
+
+        self.gain = gain
+        self.beta = beta
+        self.window_length = window_length
+        self.window = None
+        self.method = method
+        self.window_length = next_power_of_2(window_length)
+
+    def window_design(self, window_length=None, beta=None):
+        """Kaiser window design
+
+        Args:
+            window_length: Length of the window in number of samples
+            beta: Beta value for Kaiser window design
+
+        Returns:
+            window: Window designed using the beta and length provided as inputs
+
+        """
+
+        if window_length is None:
+            window_length = self.window_length
+        else:
+            self.window_length = window_length
+
+        if beta is None:
+            beta = self.beta
+        else:
+            self.beta = beta
+
+        self.window = np.kaiser(window_length, beta)
+
+        return self.window
+
+    def fit(self, X, method=None, gain=None, window_length=None, beta=None):
+        """Defines a spectral mask based on training data
+
+        Args:
+            X: Training data
+            method: Method used to calculate the Spectral Mask: 'std_dev' uses the standard
+              deviation of each frequency component and 'min_max' uses the minimum and maximum
+              values
+            gain: Multiplication factor used in the comparison between the spectral mask defined
+              using the training data and the Fourier transform of the telemetry data vector
+            window_length: Length of the sliding window in number of samples
+            beta: Beta value for Kaiser window design
+
+        """
+
+        training_signal = X
+
+        if gain is None:
+            gain = self.gain
+        else:
+            self.gain = gain
+
+        if window_length is None:
+            window_length = self.window_length
+        else:
+            self.window_length = window_length
+
+        if beta is None:
+            beta = self.beta
+        else:
+            self.beta = beta
+
+        if method is None:
+            method = self.method
+        else:
+            self.method = method
+
+        self.window_design(window_length, beta)
+
+        if method == 'std_dev':
+            self.fit_freq_std_dev(training_signal)
+        else:
+            self.fit_freq_min_max(training_signal)
+
+    def fit_freq_min_max(self, training_signal):
+        """Defines a spectral mask based on training data using min and max values of each
+             frequency component
+
+        Args:
+            training_signal: Training data
+
+        """
+
+        window_length = len(self.window)
+        window_weight = sum(self.window)
+        max_mask = np.zeros(int(window_length / 2) + 1)
+        min_mask = np.zeros(int(window_length / 2) + 1)
+
+        for i in range(0, len(training_signal) - window_length - 1):
+            temp = np.abs(np.fft.rfft(training_signal[i:i + window_length] * self.window)) \

```

Comment:
Instead of using `\` for the blank likes, consider taking advantage of the parenthesis:

```
temp = np.abs(np.fft.rfft(
    training_signal[i:i + window_length] * self.window)) / window_weight
```

or using an intermediate step:

```
rfft = np.fft.rfft(training_signal[i:i + window_length] * self.window)
temp = np.abs(rfft) / window_weight
```

Reply:
ok, I will use the second option

Reply:
Thanks! I also think the intermediate variable is the most readable option, as allows the reader to go over the code in smaller steps, and it's easier to follow :+1: